.github/actions/feature-checkout/action.yml

```
@@ -0,0 +1,53 @@
+name: 'Checkout for features'
+description: 'Checkout refs for feature deployments'
+inputs:
+  github_token:
+    description: 'A Github PAT'
+    required: true
```

Not really a Personal Access Token (PAT), just "the GitHub token".
```suggestion
  github_token:
    description: 'The GitHub token'
    required: true
```